### PR TITLE
OCPBUGS-1805: multi-zone installs may not provision storage

### PR DESCRIPTION
**Pull request by rvanderp3**
depending on which datacenter storage is being provisioned, the CSI driver may be checking if the VM is in the wrong datacenter.

### CODE REVIEW

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 1]
```diff
@@ -115,10 +116,19 @@ func (c TargetConfigController) sync(ctx context.Context, syncContext factory.Sy
...
 
 	cmString := string(c.manifest)
+
+	virtualCenterIPs := sets.StringKeySet(cfg.VirtualCenter)
+
+	if len(virtualCenterIPs) > 1 || len(virtualCenterIPs) == 0 {
+		return fmt.Errorf("cloud config %s/%s must define a single VirtualCenter", cloudConfigNamespace, cloudConfig.Name)
+	}
+
+	virtualCenterIP := virtualCenterIPs.List()[0]
```
Comment (dobsonj) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:126]: @gnufied @bertinatto Do you see a way for us to actually support multiple datacenters? Or is this the best option we have for now (throw a clear error)?

Comment (rvanderp3) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go:126]: The restriction here is actually a single vCenter.  With multi-zone support, it is very possible a customer may have multiple datacenters.

Comment (rvanderp3) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go:126]: and that could probably just be refactored to `if len(virtualCenterIPs) != 1` . :facepalm:

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 2]
```diff
@@ -133,6 +134,25 @@ func (c TargetConfigController) sync(ctx context.Context, syncContext factory.Sy
...
+
+	for pattern, value := range map[string]string{
+		"${CLUSTER_ID}":  infra.Status.InfrastructureName,
+		"${VCENTER}":     virtualCenterIP,
+		"${DATACENTERS}": cfg.VirtualCenter[virtualCenterIP].Datacenters,
+	} {
+		cmString = strings.ReplaceAll(cmString, pattern, value)
+	}
+
+	requiredCM := resourceread.ReadConfigMapV1OrDie([]byte(cmString))
```
Comment (gnufied) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:155]: This is making change in outdated code. Can you check code again and update?

Comment (rvanderp3) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go:155]: hi @gnufied odd, i just rebased

Comment (rvanderp3) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go:155]: i'll check again

**Comment by rvanderp3** [comment]
/hold

placing a hold while performing additional multi-zone testing

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go (hunk 1 above)
Comment (gnufied) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go]: So we traditionally picked this from workspace and now we are changing it to vcenter section. Could we break anyone? Such as UPI installs or people who have modified their cloud-configs? Shouldn't workspace datacenter and `VirtualCenter` datacenter be the same?

**Pull request closed without merging**